app/controllers/case_contacts_controller.rb

```diff
@@ -88,12 +88,16 @@ def edit
   end
 
   def update
+    # binding.pry
     authorize @case_contact
     @casa_cases = [@case_contact.casa_case]
     @selected_cases = @casa_cases
     @current_organization_groups = current_organization.contact_type_groups
 
     if @case_contact.update_cleaning_contact_types(update_case_contact_params)
+      additional_expense_params.each { |single_additional_expense_params|
+        @case_contact.additional_expenses.create(single_additional_expense_params)
```

Comment:
this is the cool new bug we added where it creates a new additional expense every time you try to EDIT it. Let's fix this 😂

Reply:
we can make `additional_expense_params` allow `id` and then use `update_or_create_by`

Reply:
Ok, I see the issue and I remember we talked about it last week.  This isn't working so far, but my initial approach is something like:
```
  def create_case_contact_for_every_selected_casa_case(selected_cases)
    selected_cases.map { |casa_case|
      ActiveRecord::Base.transaction do
        case_contact = casa_case.case_contacts.create(create_case_contact_params)
        if case_contact.persisted? && additional_expense_params&.any?
          additional_expense_params.each { |single_additional_expense_params|
            additional_expense_params = AdditionalExpense.exists?(id: additional_expense_params.dig("id")) ? AdditionalExpense.where(id: additional_expense_params.dig("id")).update(single_additional_expense_params) : AdditionalExpense.create(single_additional_expense_params)
            # case_contact.additional_expenses.create(single_additional_expense_params)
          }
        end
        case_contact
      end
    }
  end
```
Please let me know if you see something that makes it a bad idea (and, of course, it isn't currently working :-) )

Reply:
Deeply nested but functional, let's try it
And let's write tests :)

Reply:
I'm not finding an update_or_create method.  I did find a first_or_create that can be chained to an update.  However (just FYI) I put a binding.pry in that each block and it never triggers.  It seems its not going through that block at all.  So I'm scratching my head on that and poking around with it.

Reply:
@DrewAPeterson7671 the method is https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/find_or_create_by